modules/howtos/pages/distributed-acid-transactions-from-the-sdk.adoc

```diff
@@ -1,13 +1,455 @@
+Just remember that since the lambda may be called multiple times, so may the method.
+
+[source,python]
+----
+include::example$transactions_example.py[tag=query_examples_complex,indent=0]
+----
+
+=== Read Your Own Writes
+
+As with Key-Value operations, N1QL queries support Read Your Own Writes.
+
+This example shows inserting a document and then selecting it again.
+
+[source,python]
+----
+include::example$transactions_example.py[tag=query_insert,indent=0]
+----
+
+<1> The inserted document is only staged at this point, as the transaction has not yet committed.
+Other transactions, and other non-transactional actors, will not be able to see this staged insert yet.
+<2> But the SELECT can, as we are reading a mutation staged inside the same transaction.
+
+=== Mixing Key-Value and N1QL
+
+Key-Value operations and queries can be freely intermixed, and will interact with each other as you would expect.
+
+In this example we insert a document with Key-Value, and read it with a SELECT.
+
+[source,python]
+----
+include::example$transactions_example.py[tag=query_ryow,indent=0]
+----
+
+<1> As with the 'Read Your Own Writes' example, here the insert is only staged, and so it is not visible to other transactions or non-transactional actors.
+<2> But the SELECT can view it, as the insert was in the same transaction.
+
+=== Query Options
+
+Query options can be provided via `TransactionQueryOptions`, which provides a subset of the options in the Python SDK's `QueryOptions`.
+
+[source,python]
+----
+include::example$transactions_example.py[tag=query_options,indent=0]
+----
+
+Some of the supported options are:
+
+* scan_consistency 
+* client_context_id
+* scan_wait
+* scan_cap
+* pipeline_batch
+* pipeline_cap
+* profile
+* read_only
+* adhoc
+* raw
+
+See the xref:howtos:n1ql-queries-with-sdk.adoc#query-options[QueryOptions documentation] for details on these.
+
+include::{version-server}@sdk:shared:partial$acid-transactions.adoc[tag=query-perf]
+
+include::{version-server}@sdk:shared:partial$acid-transactions.adoc[tag=query-single]
+
+[source,python]
+----
+include::example$transactions_example.py[tag=query_single,indent=0]
+----
+
+// TODO: Unclear whether it is possible to use the `Scope` option from TransactionQueryOptions, which doesn't seem to work at this point in time.
+// Commenting out this section until we have clarification.
+//
+// You can also run a single query transaction against a particular `Scope` (these examples will exclude the full error handling for brevity):
+//
+// [source,python]
+// ----
+// include::example$transactions_example.py[tag=query_single_scoped,indent=0]
+// ----
+//
+// and configure it:
+//
+// [source,python]
+// ----
+// include::example$transactions_example.py[tag=query_single_configured,indent=0]
+// ----
+
+include::{version-server}@sdk:shared:partial$acid-transactions.adoc[tag=n1ql-rbac]
+
+== Committing
+
+// TODO: Can't find ctx.commit() for Python SDK, is this relevant?
```

Comment:
@thejcfactor just double checking, I couldn't find this anywhere, so I didn't mention it - is this relevant for Python SDK?

Reply:
No commit() in transactions.   The commit happens automatically after the transaction logic completes (or it will rollback automatically if there's an error).